cli_tools/common/utils/logging/service/cli_tool_with_log.go

```diff
@@ -0,0 +1,44 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package service
+
+import (
+	"flag"
+	"os"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+// CliToolWithLogging abstracts the interface of a cli tool with logging
+type CliToolWithLogging interface {
+	// ActionType indicates the action type of the tool
+	ActionType() ActionType
+
+	// InitParamLog initialize tool params in the log
```

Comment:
What does it mean "initialize tool params in the log"?

Reply:
modified